File: JavaScript/src/Internals/Functions/MathTrigonometric/Function_TANH.js
import { Function_1 } from '../Function_1.js';
import { Operand } from '../../../Operand.js';

class Function_TANH extends Function_1 {
    get Name() {
        return "Tanh";
    }

    constructor(a) {
        super(a);
    }

    evaluate(work, tempParameter) {
        let args1 = this.getNumber_1(work, tempParameter);
        if (args1.IsError) { return args1; }
        let x = args1.DoubleValue;
        if (x >= 66) { return Operand.Create(1); }
        if (x <= -66) { return Operand.Create(-1); }
        return Operand.Create(Math.tanh(x));
    }
}

export { Function_TANH };

File: java/src/main/java/toolgood/algorithm/internals/functions/mathtrigonometric/Function_TANH.java
package toolgood.algorithm.internals.functions.mathtrigonometric;

import java.lang.StringBuilder;

import toolgood.algorithm.Operand;
import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.internals.functions.Function_1;
import toolgood.algorithm.internals.functions.FunctionBase;

class Function_TANH extends Function_1 {
    public Function_TANH(FunctionBase func1) {
        super(func1);
    }

    @Override
    public Operand Evaluate(AlgorithmEngine work, BiFunction<AlgorithmEngine, String, Operand> tempParameter) {
        Operand args1 = func1.Evaluate(work, tempParameter);
        if (!args1.IsNumber()) {
            args1 = args1.ToNumber("Function '{0}' parameter is error!", "Tanh");
            if (args1.IsError()) {
                return args1;
            }
        }
        return Operand.Create(Math.tanh(args1.DoubleValue
()));
    }

    @Override
    public void toString(StringBuilder stringBuilder, boolean addBrackets) {
        AddFunction(stringBuilder, "Tanh");
    }
}
